src/shared/cloudformation/templateRegistry.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import { fileURLToPath, pathToFileURL } from 'url'
+import * as vscode from 'vscode'
+import { FileWatcherListener } from '../utilities/fileSystemWatcher'
+import { CloudFormation } from './cloudformation'
+
+let templateRegistry: CloudFormationTemplateRegistry | undefined
+
+export function getTemplateRegistry(): CloudFormationTemplateRegistry {
+    if (!templateRegistry) {
+        throw new Error(
+            'Template Registry not initialized. Extension code should call activate() from shared/cloudformation/activation, test code should call setTemplateRegistry().'
+        )
+    }
+
+    return templateRegistry
+}
+
+export function setTemplateRegistry(registry: CloudFormationTemplateRegistry | undefined): void {
+    templateRegistry = registry
+}
+
+export interface CloudFormationTemplateRegistry {
+    registeredTemplates: Map<string, CloudFormation.Template>
+    getRegisteredTemplate(templatePath: string): CloudFormation.Template | undefined
+    addTemplateToTemplateData(templatePath: vscode.Uri): Promise<void>
+    removeTemplateFromRegistry(templatePath: vscode.Uri): void
+}
+
+export class DefaultCloudFormationTemplateRegistryListener implements FileWatcherListener {
+    public constructor(private readonly registry: CloudFormationTemplateRegistry) {}
+
+    /**
+     * Function used by a vscode.FileSystemWatcher to handle an onDidChange event.
+     * Adds template to data map. TODO: Emit event
```

Comment:
These two lines of comments add no value. They can be replaced with a more general "updates the registry with changes made to a file", or removed altogether.

This class will not emit events. In the future, the registry itself may have change notification events, that other components can listen to. The TODO comments are not necessary either way, as that would be part of a separate implementation change to introduce more functionality.

Comment applies to all onXxx methods in this class.